### PR TITLE
Fix for the freeze issue and CTRL+L

### PR DESCRIPTION
**Pull request by syfares**

### CODE REVIEW

File: src/masternodeconfig.cpp [hunk 1]
```diff
@@ -16,41 +16,42 @@ bool CMasternodeConfig::read(boost::filesystem::path path) {
...
-        std::string alias, ip, privKey, txHash, outputIndex;
-        iss.str(line);
-        iss.clear();
-        if (!(iss >> alias >> ip >> privKey >> txHash >> outputIndex)) {
-            LogPrintf("Could not parse masternode.conf line: %s\n", line.c_str());
-            streamConfig.close();
-            return false;
-        }
-
-        if(Params().NetworkID() == CChainParams::MAIN){
```
Comment (Infernoman) [inline src/masternodeconfig.cpp]: this is the part I was speaking about, which checks the port. one more to check that it is on an IPV4 ;)

File: src/masternodeconfig.cpp [hunk 2]
```diff
@@ -16,41 +16,42 @@ bool CMasternodeConfig::read(boost::filesystem::path path) {
...
-        } else if(CService(ip).GetPort() == 17170) {
-            LogPrintf("Invalid port detected in masternode.conf: %s (17170 must be only on mainnet)\n", line.c_str());
-            streamConfig.close();
-            return false;
-        }
-
-        if (!(CService(ip).IsIPv4() && CService(ip).IsRoutable())) {
-            LogPrintf("Invalid Address detected in masternode.conf: %s (IPV4 ONLY) \n", line.c_str());
-            streamConfig.close();
-            return false;
```
Comment (Infernoman) [inline src/masternodeconfig.cpp]: and lines 41-50 through here, which checks that the IP in the masternode.conf file is IPV4 and is routable.

**Review by Infernoman — CHANGES_REQUESTED** [review]

**Comment by syfares** [comment]
Right , I have updated the file as needed

**Comment by Infernoman** [comment]
Close ;) the daemon builds successfully on linux with travis but the QT does not. just wanted to say great work though. i was a bit more inexperienced when i was working on transfercoin. so thank you for taking the time to make these changes

**Comment by syfares** [comment]
Thanks, tried a new version based on my current directory for compilation
Lots of change especially on secp256k1

**Comment by syfares** [comment]
Still no good, I think commiting from my MAC did not help, will have to stick with the windows build to keep all library correct

**Comment by syfares** [comment]
Looks better :)

**Review by Infernoman — APPROVED** [review]

**Comment by Infernoman** [comment]
great news :) im going to get in touch with the guys on the team and see if there is anything else they would like to do before an official release. I'll get this merged in now.

**Pull request merged**